default.nix

```diff
@@ -108,31 +117,30 @@ let
           inherit nix;
         };
       } ''
-      export NIX_REMOTE=local?root=$PWD
+      export NIX_REMOTE=local?root="$PWD"
 
       # A user is required by nix
       # https://github.com/NixOS/nix/blob/9348f9291e5d9e4ba3c4347ea1b235640f54fd79/src/libutil/util.cc#L478
       export USER=nobody
-      ${buildPackages.nix}/bin/nix-store --load-db < ${closure}/registration
+      ${escapeShellArg buildPackages.nix}/bin/nix-store --load-db < ${escapeShellArg closure}/registration
 
-      mkdir -p nix/var/nix/profiles nix/var/nix/gcroots
-      ln -s ${profile} nix/var/nix/profiles/default
-      ln -s ${profile} nix/var/nix/profiles/system
+      ln -s ${escapeShellArg systemProfile} nix/var/nix/profiles/default
+      ln -s ${escapeShellArg systemProfile} nix/var/nix/profiles/system
       chmod -R 755 nix/var/nix/profiles/per-user
 
-      for path in $(cat ${closure}/store-paths); do
-        cp -va $path nix/store/
-      done
+      while IFS= read -r path; do
+        cp -va "$path" nix/store/
+      done < ${escapeShellArg closure}/store-paths
 
       # Create a tarball with the Nix store for bootstraping
-      XZ_OPT="-T$NIX_BUILD_CORES" tar --owner=0 --group=0 --lzma -c -p -f $out nix
+      XZ_OPT="-T$NIX_BUILD_CORES" tar --owner=0 --group=0 --lzma -c -p -f "$out" nix
```

Comment:
We also just had this pull request that switched doing compression with a single thread: https://github.com/nix-community/nix-installers/pull/51/files#diff-245563af9e34754739b12b294f3324c1d42c116ace7ce8735e21dfbc82d8e064R132